hdc/algo/accessors.py

```diff
@@ -134,6 +136,26 @@ class DekadPeriod(Period):
     _period_cls = Dekad
 
 
+@xarray.register_dataset_accessor("season")
+@xarray.register_dataarray_accessor("season")
+class SeasonPeriod(Period):
```

Comment:
I don't think we will be able to inherit from `Period` as those objects map time onto a global and mutually exclusive set of segments, while Seasons don't. (ie 2024/10/7 will always be 202410d1, but it will be in different seasons depending on what is the relevant season range). 

In practical terms, all the properties in the `Period` class won't work for seasons, as they rely on a `_period_cls` that can be instantiated with out additional arguments; but for Season we need to pass the season_range.

 We can inherit from `AccessorTimeBase` though.